internal/errors/error.go

```diff
@@ -0,0 +1,80 @@
+package errors
+
+type (
+	// Error augments the error interface with helpers for our wrapped errors
+	Error interface {
+		error
+		Cause() error
+		Latest() error
+	}
+	// fError is a fundamental error with just a message
+	// TODO(geoah): consider adding a stack as pkg/errors does
+	fError struct {
+		message string
+	}
+	// wError is a wrapper to allow nesting errors
+	// Holds two errors, a latest and a cause
+	wError struct {
+		error error
+		cause error
+	}
+)
+
+// Error prints the message error, appended with the underlying cause
+func (e fError) Error() string {
```

Comment:
I feel like the `Error() string` functions would be clearer if they were `Message() string` or `ToString() string` instead. Cause with a name of `Error`, I would instinctively expect the method to return an actual `error` type. But again, maybe that's just me and I'm an idiot :)

Reply:
Also, shouldn't the `Error() string` or whatever it's called if my suggestion is taken seriously, be part of the `Error` interface?

Reply:
The `error` type in golang is an interface with `Error() string`. `error` does not have any "actual" form. Anything that has a `Error() string` function is an `error.

ie `https://github.com/pkg/errors`, `github.com/golang/xerrors`, etc

Reply:
Right, duh me! >_<